tests/improver-blend-adjacent-points/04-time_bounds.bats

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bats
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+. $IMPROVER_DIR/tests/lib/utils
+
+@test "weighted-blending time bounds" {
+  improver_check_skip_acceptance
+  KGO="blend-adjacent-points/time_bounds/kgo.nc"
+
+  # Run triangular time blending with matched time bounds and check it passes
+  run improver blend-adjacent-points 'forecast_period' '4' --units 'hours' \
+      --width 2 'weighted_mean' \
+      "$IMPROVER_ACC_TEST_DIR/blend-adjacent-points/time_bounds/*wind_gust*.nc" \
+      "$TEST_DIR/output.nc"
+  [[ "$status" -eq 0 ]]
+
+  improver_check_recreate_kgo "output.nc" $KGO
+
+  # Run nccmp to compare the output and kgo.
+  improver_compare_output "$TEST_DIR/output.nc" \
+      "$IMPROVER_ACC_TEST_DIR/$KGO"
+}
```

Comment:
This looks fine, however the files are currently far too large. We do not need 51 thresholds, particularly given that most are all ones or zeros. Can we please recreate the inputs with just 2 thresholds (threshold indices 9 & 10 contain non binary data), and then use these to recreate the KGO. Thanks.

Reply:
Additionally, could we have a CLI test where we attempt to merge 3 hour and 1 hour accumulation data for example and show that this fails.

Reply:
OK this is a problem.
It doesn't fail.
The reason it doesn't fail is that we don't actually blend over "time" in any of these cases from the CLI / suite app.  We blend over forecast period.  The reason for THIS is that we need to specify a central point as input to the CLI, rather than determining it automatically, and we don't want to pass in a numerical time point - so we use forecast period as a proxy.
There are a number of ways to hack this, and no obvious way to fix it.  I'll put some more details on the JIRA ticket.  But this requires some more discussion before it can be merged.

Reply:
Is this sorted now?

Reply:
Yes - this is what the new argument and suite PR is for.